tests/testthat/test_prepareAnnotations.R

```diff
@@ -1,37 +1,169 @@
+    gr <- readRDS(test_path("fixtures", "grGTF.rds"))
+    
     expectedGR <- readRDS(system.file("extdata", "annotationGranges_txdbGrch38_91_chr9_1_1000000.rds", package = "bambu"))
+  
+    expect_named(mcols(gr), c("TXNAME", "GENEID", "txid", "eqClass", "eqClassById"))
+    expect_equal(mcols(gr), mcols(expectedGR))
+})
 
-    gr <- prepareAnnotations(x = txdb)
 
-    expect_equal(gr, expectedGR)
-    expect_s4_class(gr, class = "CompressedGRangesList")
-    expect_named(mcols(gr), c("TXNAME", "GENEID", "eqClass"))
+test_that("positive strand gives ascending exon_rank", {
+    gr <- readRDS(test_path("fixtures", "grGTF.rds"))
+    
+    unlisted_gr <- unlist(gr)
+    
+    positive_check <- data.frame(txname = names(unlisted_gr), unlisted_gr) %>% 
+        filter(strand == "+") %>% 
+        group_by(txname) %>% 
+        summarise(validate = all(exon_rank == seq(n())))
+    
+    expect_true(all(positive_check$validate))
 })
 
 
-test_that("prepareAnnotations of genome library is a GRangesList", {
-    library(TxDb.Hsapiens.UCSC.hg38.knownGene)
-    gr <- prepareAnnotations(TxDb.Hsapiens.UCSC.hg38.knownGene)
+test_that("positive strand gives descending exon_endRank", {
+    gr <- readRDS(test_path("fixtures", "grGTF.rds"))
+    
+    unlisted_gr <- unlist(gr)
+    
+    positive_check <- data.frame(txname = names(unlisted_gr), unlisted_gr) %>% 
+        filter(strand == "+") %>% 
+        group_by(txname) %>% 
+        summarise(validate = all(exon_endRank == seq(n(),1)))
+  
+    expect_true(all(positive_check$validate))
+})
+
 
-    expect_s4_class(gr, class = "CompressedGRangesList")
-    expect_named(mcols(gr), c("TXNAME", "GENEID", "eqClass"))
+test_that("negative strand gives descending exon_rank", {
+    gr <- readRDS(test_path("fixtures", "grGTF.rds"))
+    
+    unlisted_gr <- unlist(gr)
+    
+    negative_check <- data.frame(txname = names(unlisted_gr), unlisted_gr) %>% 
+        filter(strand == "-") %>% 
+        group_by(txname) %>% 
+        summarise(validate = all(exon_rank == seq(n(),1)))
+  
+    expect_true(all(negative_check$validate))
 })
 
 
+test_that("negative strand gives ascending exon_endRank", {
+    gr <- readRDS(test_path("fixtures", "grGTF.rds"))
+    
+    unlisted_gr <- unlist(gr)
+    
+    negative_check <- data.frame(txname = names(unlisted_gr), unlisted_gr) %>% 
+        filter(strand == "-") %>% 
+        group_by(txname) %>% 
+        summarise(validate = all(exon_endRank == seq(n())))
+  
+    expect_true(all(negative_check$validate))
+})
 
 
-test_that("prepareAnnotations from gtf file is GRangesList", {
-    gtf.file <- system.file("extdata", "Homo_sapiens.GRCh38.91_chr9_1_1000000.gtf", package = "bambu")
+test_that("txid must be in EqClassById", {
+    gr <- readRDS(test_path("fixtures", "grGTF.rds"))
+    
+    check <- data.frame(mcols(gr)) %>% 
+        dplyr::select(txid, eqClassById) %>% 
+        rowwise() %>% 
+        mutate(validate = txid %in% eqClassById)
```

Comment:
```suggestion
        mutate(validate = txid %in% unlist(eqClassById))
```

Reply:
resolved in f3347e641bb0020e84d75503da766cb5dd681ab4